app/src/main/java/com/example/countryinfo/CountryDetailsAdapter.kt

```diff
@@ -64,38 +52,43 @@ class CountryDetailsAdapter(
             with(itemView) {
                 listOf(titleText, titleText2, titleText3).forEachIndexed { index, textView ->
                     state.info.getOrNull(index)?.let {
+                        if (index == 1) {
+                            setMyOnClickListener(state.detailsSections, state.info)
+                        }
                         textView.apply {
                             text = it
                             setBackgroundResource(state.detailsSections.bgShapeId)
-                            isVisible = true
                         }
-                        if (index == 0) {
-                            viewSection = null
-                        } else {
-                            viewSection = state.detailsSections
-                            dialogMessage = state.info
-                            itemView.post { checkViewsWidth() }
-                        }
-                    } ?: run { textView.isVisible = false }
+                    }
                 }
+                itemView.post { checkViewsWidth() }
             }
         }
 
         private fun checkViewsWidth() {
             val boundsWidth =
-                itemView.titlesBounds.measuredWidth - itemView.context.resources.getDimension(R.dimen.titles_margin)
-                    .toInt()
-            val view1 = itemView.titleText.measuredWidth
-            val view2 = itemView.titleText2.measuredWidth
-            val view3 = itemView.titleText3.measuredWidth
-            if (view1 + view2 > boundsWidth) {
-                itemView.titleText2.isVisible = false
-                itemView.titleText3.isVisible = false
-                itemView.truncateDots.isVisible = true
-                return
-            } else if (view1 + view2 + view3 > boundsWidth) {
-                itemView.titleText3.isVisible = false
-                itemView.truncateDots.isVisible = true
+                itemView.titlesBounds.measuredWidth -
+                        (itemView.titlesBounds.layoutParams as ViewGroup.MarginLayoutParams).marginStart
```

Comment:
[nit] Please check if we have some extension for this?

Reply:
I haven't found any info on that.